cli-plugins/apps/how-to-guides.hbs.md

```diff
@@ -0,0 +1,611 @@
+            2 + |apiVersion: carto.run/v1alpha1
+            3 + |kind: Workload
+            4 + |metadata:
+            5 + |  name: myworkload
+            6 + |  namespace: default
+            7 + |spec:
+            8 + |  source:
+            9 + |    image: gcr.io/my-registry/my-workload-image:latest@sha256:f28c5fedd0e902800e6df9605ce5e20a8e835df9e87b1a0aa256666ea179fc3f
+           10 + |    subPath: subpath_folder
+      ❓ Do you want to create this workload? [yN]:
+
+      ```
+
+## <a id='tanzuignore-file-usage'> .tanzuignore file
+
+As more systems around us adopt the "as code" approaches, application developers will increasingly have files in their projects that have nothing to do with actually running code (those files don't end up in the running container).
+
+When creating a workload from local source code, these unused files can be added to the `.tanzuignore` file so there won't be an unnecesary consumption of resources and also, the workload deployment won't be triggered when one of those files changes. The file must contain itself as well for it to be ignored.
+
+The `.tanzuignore` file must be located at the root of the project so Apps CLI plug-in can read it.
+It allows to list folders (these must not end with the system separator, e.g. `/` or `\`), files and have comments which start with `#`. If the file contains files or directories that are not in the source code, they are ignored.
+
+**Example of a .tanzuignore file**
+```bash
+    .tanzuignore # must contain itself in order to be ignored
+    # This is a comment
+    this/is/a/folder/to/exclude
+
+    this-is-a-file.ext
+```
+
+<!-- ## <a id='maximize-efficiency'> Leveraging ENVs to maximize efficiency
+
+## <a id='param-yaml-usage'> --param-yaml
+
+## <a id='app-usage'> --app
+
+## <a id='type-usage'> --type
+
+## <a id='labels-anns-env'> --labels, --annotations, --env, --build-env -->
+
+## <a id='dry-run-usage'> --dry-run
+
+The main goal of `--dry-run` flag is to prepare all the steps to submit a workload to the cluster and stop before sending it, showing
+an output of the final structure of the workload.
+
+For example, when applying a workload from Git source:
+
+```bash
+tanzu apps workload apply rmq-app --git-repo https://github.com/jhvhs/rabbitmq-sample --git-branch main --service-ref "rmq=rabbitmq.com/v1beta1:RabbitmqCluster:example-rabbitmq-cluster-1" -t web --dry-run
+---
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  creationTimestamp: null
+  labels:
+    apps.tanzu.vmware.com/workload-type: web
+  name: rmq-app
+  namespace: default
+spec:
+  serviceClaims:
+  - name: rmq
+    ref:
+      apiVersion: rabbitmq.com/v1beta1
+      kind: RabbitmqCluster
+      name: example-rabbitmq-cluster-1
+  source:
+    git:
+      ref:
+        branch: main
+      url: https://github.com/jhvhs/rabbitmq-sample
+status:
+  supplyChainRef: {}
+```
+
+This will allow the user to check how it's workload would be created/updated in the cluster.
```

Comment:
This will allow the user to check how a workload _would be_ created/updated in the cluster based on the current specifications passed in via `--file workload.yaml` and/or command flags.